pkg/bundle/sync.go

```diff
@@ -148,8 +163,19 @@ func (b *bundle) syncTarget(ctx context.Context, log logr.Logger, bundle *trusta
 		return false, fmt.Errorf("failed to get configmap %s/%s: %w", namespace, bundle.Name, err)
 	}
 
-	var needsUpdate bool
+	// Here, the config map exists, but the selector doesn't match the namespace.
+	if !matchNamespace {
+		// The ConfigMap is owned by this controller- delete it.
+		if metav1.IsControlledBy(&configMap, bundle) {
+			log.V(2).Info("deleting bundle from Namespace since namespaceSelector does not match")
+			return true, b.client.Delete(ctx, &configMap)
+		}
+		// The ConfigMap isn't owned by us, so we shouldn't delete it. Return that
```

Comment:
Should there be a warning event here?

Reply:
This has now been implemented by sending an event to the ConfigMap when the ConfigMap is not deleted due to no longer being owner by trust.